src/internal/git/utils.go

```diff
@@ -28,6 +28,11 @@ type Credential struct {
 	Auth http.BasicAuth
 }
 
+var (
+	// For further explanation: https://regex101.com/library/UfILls and https://regex101.com/rary/UfILls
+	gitURLRegex = regexp.MustCompile(`^(.+)\/([\w\-\.]+?)(\.git)?(@([\w\-\.]+))?$`)
```

Comment:
Now that I know named capture groups are a thing, could you do that here?

Reply:
done